python rewrite of the simulator in order to better algin with other simulators

ianzhang366 opened the pull request:
In order to achieve the following issue, we need to make sure the simulator will respond to a `ManagedCluster` Add an event.

https://github.com/open-cluster-management/backlog/issues/13877


Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
parent d04b9db7002ae008bfacbd8afc3b35f331d4a4a9
author Ian Zhang <izhang@redhat.com> 1625252325 -0400
committer testadmin <testadmin@redhat.com> 1626277901 -0400

initial python rewrite of setup-metrics-collector.sh and setup-metrics-collector.sh

Signed-off-by: Ian Zhang <izhang@redhat.com>

use logging instead of printing to stdout

Commit:
initial python rewrite of setup-metrics-collector.sh and setup-metrics-collector.sh

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
initial python rewrite of setup-metrics-collector.sh and setup-metrics-collector.sh

Commit:
add description

Commit:
add description

Signed-off-by: Ian Zhang <izhang@redhat.com>

marcolan018 commented on test/simulator/metrics-collector-simulator.py:
```diff
@@ -0,0 +1,355 @@
+    for e in api.watch():
+        print(e['type'])
+
+        if e['type'] == "ADDED":
```
do we need to watch "DELETED" event?

ianzhang366 replied:
I was thinking for using the `--clean` flag to delete the simulators, so I didn't watch the `DELETED`

marcolan018 commented on test/simulator/metrics-collector-simulator.py:
```diff
@@ -0,0 +1,355 @@
+                        clusterName))
+            except ApiException as err:
+                print("a", err.status)
+                if err.status != 404:
```
if run into error, should we just log the error and continue to next cluster, but not return error?

ianzhang366 replied:
yup, you are right, let it fall through makes more sense when deleting

Commit:
fall through when deleting

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
fall through when deleting

Commit:
update option

Commit:
update option

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
use logging instead of printing to stdout

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
use logging instead of printing to stdout

Commit:
adding sign-off
Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
adding sign-off
Signed-off-by: Ian Zhang <izhang@redhat.com>

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
add new

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
add new

Signed-off-by: Ian Zhang <izhang@redhat.com>

Commit:
Merge branch 'main' of github.com:ianzhang366/metrics-collector

Commit:
parent d04b9db7002ae008bfacbd8afc3b35f331d4a4a9
author Ian Zhang <izhang@redhat.com> 1625252325 -0400
committer testadmin <testadmin@redhat.com> 1626277901 -0400

initial python rewrite of setup-metrics-collector.sh and setup-metrics-collector.sh

Signed-off-by: testadmin <testadmin@redhat.com>
